### PR TITLE
Fix compile on Linux

### LINKED ISSUE #50: include/vuk/ShaderSource.hpp should include cstring for memcpy

### LINKED ISSUE #52: Don't assume size_t is unsigned long long
It's unsigned long on 64-bit Linux and causes e.g. src/Allocator.cpp:492 to error

### PR DESCRIPTION
**Pull request by aleokdev**
Adds some missing includes & adds some explicit conversions.
Closes #52 & closes #50

### CODE REVIEW

File: src/Allocator.cpp [hunk 1]
```diff
@@ -488,8 +488,8 @@ namespace vuk {
 		pool_helper->result = VK_NULL_HANDLE;
 		auto mem_reqs = pool.mem_reqs;
 		mem_reqs.size = size;
-		mem_reqs.alignment = std::max(1ull, mem_reqs.alignment);
-		alignment = std::max(1ull, alignment);
+		mem_reqs.alignment = std::max((size_t)1ull, mem_reqs.alignment);
+		alignment = std::max((size_t)1ull, alignment);
```
Comment (nanokatze) [inline src/Allocator.cpp:492]: Just make it static_cast<size_t>(1) or 1uz when C++23 arrives

**Comment by kociap** [comment]
Use `static_cast<size_t>` instead of C casts.

**Comment by HaedHutner** [comment]
i would like to formally object to being muted in your discord, i have done nothing wrong

**Comment by martty** [comment]
Thanks!

**Pull request merged**